ui/src/middlewares/statePersister.js

```diff
@@ -0,0 +1,40 @@
+import { fromJS } from 'immutable';
```

Comment:
``persistentState.js`` sounds better (?)

Reply:
i don't know, we have already `searchDispatcher`.